### PR TITLE
#160618113 search functionality

### PR DESCRIPTION
**Pull request by Phunmbi**
### What does this PR do?
This pull request implements the search functionality

#### Description of Task to be completed?
- Create this route for search `api/article/search`
- test the functionality.

#### How should this be manually tested?
- Setup the repo
- run `npm install`
- run `npm run test-migrate` to automatically migrate your test database
- run `npm run test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#160618113](https://www.pivotaltracker.com/story/show/160618113)

#### Screenshots (if appropriate)
N/A

#### Questions:
None

### CODE REVIEW

**Review by MaejorAli — APPROVED** [review]
Looks good! I like that your code is well commented. well done.

**Review by VictorUkafor — APPROVED** [review]
looks good

File: tests/searchControllerTest.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+          done();
+        });
+    });
+
+    it('should search by tag', (done) => {
+      chai
+        .request(app)
+        .get('/api/articles/search?article=false&author=false&tag=angular')
+        .end((err, res) => {
+          console.log(res.body);
```
Comment (AdinoyiSadiq) [inline tests/searchControllerTest.js]: Hi. Please, could you remove the console statement here? Thanks

Comment (Phunmbi) [reply tests/searchControllerTest.js]: Done

**Review by AdinoyiSadiq — APPROVED** [review]
Looks good.

**Pull request merged**